uaclient/tests/test_ua_timer.py

```diff
@@ -2,100 +2,130 @@
-            job_name="Day Job", job_func=failed_job, cfg=cfg
-        )
-        assert "Error executing job Day Job: Something broke" in caplog_text()
+        job = TimedJob("day_job", failed_job, 14400)
+        assert False is job.run(cfg)
+        assert "Error executing job day_job: Something broke" in caplog_text()
+
+    @pytest.mark.parametrize("is_cfg_set", (False, True))
+    def test_get_default_run_interval(self, is_cfg_set, FakeConfig):
+        """Use the default run interval when not overriden."""
+        cfg = FakeConfig()
+        if is_cfg_set:
+            setattr(cfg, "day_job_timer", None)
+        job = TimedJob("day_job", lambda: None, 14400)
+
+        assert 14400 == job.run_interval_seconds(cfg)
+
+    def test_get_configured_run_interval(self, FakeConfig):
+        """Use the configured run interval when not overriden."""
+        cfg = FakeConfig()
+        setattr(cfg, "day_job_timer", 28800)
+        job = TimedJob("day_job", lambda: None, 14400)
+
+        assert 28800 == job.run_interval_seconds(cfg)
 
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    def test_does_not_run_if_disabled_in_config(self, FakeConfig, caplog_text):
+        cfg = FakeConfig()
+        setattr(cfg, "day_job_timer", 0)
+
+        def disabled_job(config):
+            assert config == cfg
+
+        job = TimedJob("day_job", disabled_job, 14400)
+        assert False is job.run(cfg)
+        assert "Not running disabled job: day_job" in caplog_text()
+
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    def test_does_not_run_if_disabled_by_default(
+        self, FakeConfig, caplog_text
+    ):
+        cfg = FakeConfig()
 
-class TestRunJobs:
-    @pytest.mark.parametrize("next_run,call_count", ((None, 1),))
-    @mock.patch("lib.timer.run_job")
+        def disabled_job(config):
+            assert config == cfg
+
+        job = TimedJob("day_job", disabled_job, 0)
+        assert False is job.run(cfg)
+        assert "Not running disabled job: day_job" in caplog_text()
+
+
+class TestTimer:
+    @pytest.mark.parametrize("has_next_run", (False, True))
     def test_run_jobs_persists_job_status_on_successful_run(
-        self, run_job, next_run, call_count, FakeConfig
+        self, has_next_run, FakeConfig
     ):
         """Successful job run results in updated job-status.json."""
         cfg = FakeConfig()
         now = datetime.datetime.utcnow()
-        if next_run:
+
+        if has_next_run:
             cfg.write_cache(
                 "jobs-status",
-                {
-                    "update_messaging": {
-                        "next_run": now - datetime.timedelta(seconds=1)
-                    },
-                    "update_status": {
-                        "next_run": now - datetime.timedelta(seconds=1)
-                    },
-                },
+                {"day_job": {"next_run": now - datetime.timedelta(seconds=1)}},
```

Comment:
1. As this is a unit test, a generic entry here  tests only the functionality of this piece of code, without depending on the real environment around it
2. Follows from 1: when any new job is added, there is no side effect, and no need to change the test because it is one more item to iterate over the same logic

Reply:
I get the idea, but I do like the idea that for every job we run, we will need to modify that test. It should at least cover that we are properly configuring the time for each job correctly. But let me know if you think this is an overkill, as I don't have a strong opinion on that

Reply:
I see. I think that to test the jobs themselves, with the real values, a behave test would make more sense  - can try and orchestrate a way of checking if next_run is being calculated appropriately.

Reply:
I think this be harder to check on the behave tests, but if you think there is an easy way to do that, go for it